### PR TITLE
A terser .gitignore file.

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
I trimmed _.gitignore_ to the stuff that looked like it was still useful (at least, to me). Most of what's in there seemed like remnants of previous build systems (Buck, Stack, etc.). Other stuff includes editor- and OS-specific configuration files; I've left in what I recognized.

I've tried to categorize the files. This helps with keeping the file trim.

On my machine, this has exactly the same effect as the longer one. However, it's highly likely that I've overlooked something; let me know and I can add it back.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by SamirTalwar-DA** [comment]
OK, which editors should we support? I see emacs and vim backup files in there, Ensime, Eclipse, and IntelliJ IDEA. However, there could be 10 more that I don't recognize.

Happy to add them back if we can identify the ones we care about. (And add in whatever was missing for those editors.)

**Comment by cocreature** [comment]
I would keep all editor-specific stuff and only remove stuff that is explicitly obsolete (e.g. references to rattle, cabal files or stack).

**Comment by cocreature** [comment]
If you don’t recognize something, it’s probably reasonable to remove it I guess.

**Comment by SamirTalwar-DA** [comment]
I added back in editor files that I recognized. Hopefully that's good enough for now. We can always add more.

File: .gitignore [hunk 1]
```diff
@@ -18,11 +18,44 @@
...
+# IntelliJ IDEA
+.idea
+.ijwb # Bazel
+*.iml
+*.ipr
+*.iws
+out
+
+# Vim
+*~
```
Comment (cocreature) [inline .gitignore:58]: This is also emacs!

Comment (SamirTalwar-DA) [reply .gitignore:58]: Haha! I need to add the Vim swap files back in, too.

Comment (SamirTalwar-DA) [reply .gitignore:59]: Duplicated it for Emacs; if we delete one this should still work.

**Pull request merged**